### PR TITLE
Add ERC20 Coins 

### PR DESCRIPTION
**Pull request by raph0792**

### CODE REVIEW

**Comment by raph0792** [comment]
HBX 365 and Voltz coins

File: server_extensions_extra/src/main/resources/batm-extensions.xml [hunk 1]
```diff
@@ -189,6 +189,25 @@
         <cryptologo cryptocurrency="BTC" file="btc.svg"/>
     </extension>
     <extension class="com.generalbytes.batm.server.extensions.extra.bitcoincash.BitcoinCashExtension">
+        </exchange>
```
Comment (rpanak-generalbytes) [inline server_extensions_extra/src/main/resources/batm-extensions.xml]: Invalid XML

File: server_extensions_extra/src/main/resources/batm-extensions.xml [hunk 2]
```diff
@@ -189,6 +189,25 @@
...
+                <param name="userId" />
+                <param name="accountId" />
+                <param name="apikey" />
+                <param name="apisecret" />
+                <param name="fiatcurrency" />
+                <cryptocurrency>Voltz</cryptocurrency>
+                <cryptocurrency>365</cryptocurrency>
+        </exchange>
+  </extension>
+      <extension class="com.generalbytes.batm.server.extensions.extra.bitcoincash.BitcoinCashExtension">
```
Comment (rpanak-generalbytes) [inline server_extensions_extra/src/main/resources/batm-extensions.xml]: You are defining the previous extension again. Not cool!

File: server_extensions_extra/src/main/resources/batm-extensions.xml [hunk 3]
```diff
@@ -883,6 +902,47 @@
         </wallet>
         <cryptologo cryptocurrency="REP" file="rep.png"/>
         <cryptologo cryptocurrency="REP" file="rep.svg"/>
+<extension class="com.generalbytes.batm.server.extensions.extra.ethereum.EthereumExtension" >
```
Comment (rpanak-generalbytes) [inline server_extensions_extra/src/main/resources/batm-extensions.xml]: Why are you redefining the Ethereum extension again?

File: server_extensions_extra/src/main/resources/batm-extensions.xml [hunk 4]
```diff
@@ -883,6 +902,47 @@
...
+            <!-- Hint: infuraERC20_TOKENSYMBOL_DECIMALPLACES_CONTRACTADDRESS -->
+            <param name="apiKey" />
+            <param name="mnemonic" />
+            <help>Mnemonic should be 24 words delimited by spaces</help>
+            <cryptocurrency>HBX</cryptocurrency>
+        </wallet>
+        <cryptologo cryptocurrency="HBX" file="hbx.png"/>
+        <cryptologo cryptocurrency="HBX" file="hbx.svg"/>
+         </extension>
+<extension class="com.generalbytes.batm.server.extensions.extra.ethereum.EthereumExtension" >
```
Comment (rpanak-generalbytes) [inline server_extensions_extra/src/main/resources/batm-extensions.xml]: Why are you redefining the Ethereum extension again?

Comment (rpanak-generalbytes) [reply server_extensions_extra/src/main/resources/batm-extensions.xml]: Also make sure the PNG and SVG files are present.

**Comment by rpanak-generalbytes** [comment]
I guess you wanted to add support for some ERC20 tokens. I'd suggest taking a look at how DAI or BAT tokens are implemented (`batm-extensions.xml`). You need a wallet, rate source and logos. The rate sources implementation may require changes outside of the `batm-extensions.xml` file.

**Comment by rpanak-generalbytes** [comment]
Have you successfully tested your changes? See the main repository page for information on how to run the tester.

**Comment by raph0792** [comment]
Yeah, I think it is okay now. Can you please merge it ?
How long does it takes for it to apply on our machine ???

**Comment by rpanak-generalbytes** [comment]
Is it a yes or no???

Have you successfully tested your changes? Yes or no answer, please.

**Comment by raph0792** [comment]
yes

**Comment by raph0792** [comment]
When will it get merged ???

**Comment by raph0792** [comment]
> Is it a yes or no???
> 
> Have you successfully tested your changes? Yes or no answer, please.

yes

**Comment by rpanak-generalbytes** [comment]
No, you have not. Otherwise, you'd have found out the XML is not valid.

**Comment by raph0792** [comment]
Kindly tell what we have to do and what is wrong in it. We need to merge ASAP. and please understand that we never want to waste your time or interrupt all we need is help

**Comment by rudygodoy** [comment]
@raph0792 perhaps it is necessary for you to understand how a community-driven open source project works in the first place. Even though GB have created this project to receive contributions that eventually (this is important) get integrated in to the main software, it is not obligated to merge them at all, nor provide help or support. What I'd do in your situation is either hire them for support or hire a third party to get things in place.

**Comment by raph0792** [comment]
> @raph0792 perhaps it is necessary for you to understand how a community-driven open source project works in the first place. Even though GB have created this project to receive contributions that eventually (this is important) get integrated in to the main software, it is not obligated to merge them at all, nor provide help or support. What I'd do in your situation is either hire them for support or hire a third party to get things in place.

I requested for the support through mail many time. We are okay to hire generalbytes team to help us out but there were no reply from their side. I requested for help we are willingly to hire. but no one answer us and try to support us at all . what do we do now ??

**Comment by rpanak-generalbytes** [comment]
@raph0792 I have searched for your tickets with our support department and **all** of them have been answered promptly. When you asked for help with the implementation, you were given the steps needed to continue. The instructions were given in hoping that they can save you money and time if you have someone with the required skills. While I understand that for someone non-technical the instructions can be too much to handle, you seemed satisfied with the answers.
Later you have submitted the PRs and started being dishonest when asked about testing your changes first. Every PR check takes time. If you are skipping the tests you can do yourself, it is time that could be spent better on other things (eg. a different PR). I don't think there is anything rude in GB's conversation with you.

Anyway, as you mention you can't find anyone with the required skills. In such a case, we can offer you development services. I'll let our support department contact you via a private channel (probably Kayako) with more details.

**Comment by raph0792** [comment]
could you please recheck it again

**Comment by rpanak-generalbytes** [comment]
@raph0792 Have you tested the changes?

**Comment by raph0792** [comment]
yes, we did test the change . what do we have to do next

**Comment by rpanak-generalbytes** [comment]
How did you test the change and what was the output you got from the tester?

**Comment by raph0792** [comment]
Hello,

The team checked it according to what you mention in your first page and run test according to what you told and it was showing  Wallet , Exchange and rate source. what is the next step?

**Comment by rpanak-generalbytes** [comment]
The front page only shows how to run tester without any test action. Use the output of the tester to specify additional parameters on the command line and test all the stuff you have added (wallet, exchange, rate source).

**Comment by raph0792** [comment]
My team said they have done it already. what is the next steps we have to do ?

**Comment by rpanak-generalbytes** [comment]
I doubt they have done it. If they have, however, paste here the output of:
- exchange test
- wallet test
- rate source test

Include the command line used to run the test.

**Comment by raph0792** [comment]
screen shot 2562-01-24 at 22 12 54
screen shot 2562-01-24 at 22 13 20
screen shot 2562-01-24 at 22 14 08
screen shot 2562-01-24 at 22 15 01

**Comment by raph0792** [comment]
Here are all the result they got  is there anymore change do we have to change

**Comment by rpanak-generalbytes** [comment]
1) The code in this PR can't print those outputs. Are you testing code that is not part of this PR?
2) Those commands are just listing the available rate sources, exchanges and wallets. You need to test your implementations with the actions described in tester help message (i.e. the get-rates, get-wbalance and get-ebalance actions). If you are successful, please provide the output and the commands as before.

**Comment by generalbytes** [comment]
Please respond. Otherwise this PR will be closed for inactivity.

**Comment by rpanak-generalbytes** [comment]
Closing for inactivity. Feel free to reopen if necessary.

**Pull request closed without merging**